rqueue-core/src/main/java/com/github/sonus21/rqueue/core/DefaultRqueueMessageConverter.java

```diff
@@ -18,27 +18,18 @@
 
 import com.github.sonus21.rqueue.converter.GenericMessageConverter;
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
+import com.google.common.collect.ImmutableList;
+import lombok.EqualsAndHashCode;
 import org.springframework.messaging.converter.CompositeMessageConverter;
 import org.springframework.messaging.converter.MessageConverter;
 import org.springframework.messaging.converter.StringMessageConverter;
 
-public final class MessageConverterFactory {
-  private MessageConverterFactory() {}
+@EqualsAndHashCode(callSuper = true)
+public final class DefaultRqueueMessageConverter extends CompositeMessageConverter implements MessageConverter{
 
-  public static CompositeMessageConverter getMessageConverter(
```

Comment:
If we want to not instantiate many objects of message converter then we can just create object once and return the same on each method call without considering the parameters.